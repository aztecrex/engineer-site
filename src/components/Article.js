import React from 'react';
import {connect} from 'react-redux';
import R from 'ramda';
import Markdown from './Markdown';


const computeSource = (id, index, content) => {

  let source;

  const digest = R.path([id,'digest'], index);
  if (!digest)
    source="NOT FOUND";
  else {
    const loaded = R.path([digest], content);
    if (!loaded)
      source = "LOADING";
    else {
      source = loaded;
    }
  }

  return source;
}

class Article extends React.Component {

  render() {
    const source = computeSource(
      this.props.id,
      this.props.index,
      this.props.content
    );
    return (
      <Markdown source={source} />
    );
  }

};

const mapStateToProps = (state) => {

  const index = R.path(['articles','index'],state) || {};
  const content = R.path(['articles','content'],state) || {};

  return {index, content};

};

export default connect(mapStateToProps)(Article);
export {Article, mapStateToProps};
