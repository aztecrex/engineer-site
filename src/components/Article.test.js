import React from 'react';
import {shallow, render} from 'enzyme';
import Normalizer from 'html-normalizer';
const normalizer = new Normalizer();

import Article from './Article';

describe('render article', () => {

  it('renders as a div', () => {

    // given
    let article = (<Article />);

    // when
    let wrapper = shallow(article);

    // then
    expect(wrapper.is('div')).toBe(true);

  });

  it('renders the provided html', () =>  {

    // given
    let content =   "<p id='itsme'>pretty nice paragraph</p>"
                  + "\n<hr><p class='purple'>what is next</p>";
    let article = (<Article content={content} />);

    // when
    let wrapper = render(article);

    // then
    let expected = normalizer.normalize(content);

    let inner = wrapper.find('#itsme');
    let actual = normalizer.normalize(inner.parent().html());

    expect(actual).toEqual(expected);


  });

});
