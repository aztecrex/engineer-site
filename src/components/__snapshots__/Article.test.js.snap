exports[`render unconnected renders content found 1`] = `
<Markdown
  source="the content you need" />
`;

exports[`render unconnected renders content not found 1`] = `
<Markdown
  source="LOADING" />
`;

exports[`render unconnected renders id not found 1`] = `
<Markdown
  source="NOT FOUND" />
`;
